MonoGame.Framework/Graphics/TextureCollection.DirectX.cs

```diff
@@ -12,8 +12,14 @@ void PlatformInit()
 
         internal void ClearTargets(GraphicsDevice device, RenderTargetBinding[] targets)
         {
-            ClearTargets(targets, device._d3dContext.VertexShader);
-            ClearTargets(targets, device._d3dContext.PixelShader);
+            if (_applyToVertexStage)
+            {
+                ClearTargets(targets, device._d3dContext.VertexShader);
+            }
+            else
+            {
+                ClearTargets(targets, device._d3dContext.PixelShader);
+            }
```

Comment:
I agree, that would be cleaner. `PlatformInit` would be the place to do it, but the device context isn't available there, because at the time the `TextureCollection` is created, the D3D device context hasn't been created yet. So it would require some refactoring.